### PR TITLE
Unit test for encoding to string w/ checksum

### LINKED ISSUE #222: Unit test: Checksummed ascii-safe encoding

### PR DESCRIPTION
**Pull request by larowlan**
Fixes #222

### CODE REVIEW

**Comment by defuse** [comment]
Awesome! Can you move the `testBadHexEncoding` one too please? :)

**Comment by larowlan** [comment]
Sure, will do so later today

**Comment by larowlan** [comment]
Done. Are the failing tests related to PGP a known issue?
`Could not download PGP public key for verification`

File: test/unit/EncodingTest.php [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?php
 
+use Defuse\Crypto\Core;
```
Comment (paragonie-scott) [inline test/unit/EncodingTest.php]: Very slight nit: Could you add a `\` prefix here? Just for consistency. :)

**Comment by defuse** [comment]
Yeah the tests are failing because the composer key is gone from the `pgp.mit.edu` keyserver (well, sometimes it is, sometimes it isn't).

**Comment by defuse** [comment]
We might as well just check in the .asc of the composer public key.

**Comment by larowlan** [comment]
@paragonie-scott sorted - also re-ran `php-cs-fixer fix` cause some issues have crept in since #200

**Comment by larowlan** [comment]
Ohh we got a green run :)

**Comment by defuse** [comment]
@larowlan: I intentionally didn't apply php-cs-fixer to random-compat since it's a separate project -- I want the code to remain identical to upstream.Is there a way to exclude that directory in the config?

**Comment by larowlan** [comment]
Yeah, will fix shortly

**Comment by defuse** [comment]
@larowlan: Thanks, is there any progress on getting rid of the php-cs-fixer changes, or should I just cherry pick the first commit?

**Comment by defuse** [comment]
I cherry-picked the commit, thanks!

**Comment by larowlan** [comment]
@defuse - sorry I didn't get back to it...will send a new PR with updated
config

On 19 April 2016 at 10:48, Taylor Hornby <notifications@github.com> wrote:

> Closed #228 <https://github.com/defuse/php-encryption/pull/228>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/defuse/php-encryption/pull/228#event-632185965>
>



-- 
Lee Rowlands
Senior Drupal Developer
PreviousNext
skype: larowlan
email: larowlan@previousnext.com.au

**Pull request closed without merging**